croaring/src/bitmap/serialization.rs

```diff
@@ -0,0 +1,215 @@
+use super::{Bitmap, BitmapView};
+
+use std::ffi::{c_char, c_void};
+
+pub trait Serializer {
+    fn serialize_into<'a>(bitmap: &Bitmap, dst: &'a mut Vec<u8>) -> &'a [u8];
+    fn get_serialized_size_in_bytes(bitmap: &Bitmap) -> usize;
+}
+
+pub trait Deserializer {
+    fn try_deserialize(buffer: &[u8]) -> Option<Bitmap>;
+}
+
+pub trait ViewDeserializer {
+    unsafe fn deserialize_view(data: &[u8]) -> BitmapView<'_>;
+}
+
+/// The `Portable` format is meant to be compatible with other roaring bitmap libraries, such as Go or Java.
+/// It's defined here: https://github.com/RoaringBitmap/RoaringFormatSpec
+pub enum Portable {}
```

Comment:
I wonder if it might make sense to pull `Portable`, `Frozen`, and `Native` out of `crate::bitmap::serialize` into a `crate::serialize`, since they're not bitmap specific (we use the same types for treemap serialization)

Just a thought, definitely not worth holding up the PR for.